app/src/main/java/zapsolutions/zap/util/Wallet.java

```diff
@@ -194,6 +218,74 @@ public void run() {
         }
     }
 
+    /**
+     * Call this if the deamon is running, but the wallet is not unlocked yet.
+     *
+     * @param password
+     */
+    public void unlockWallet(String password) {
+        //UnlockWallet
+        WalletUnlockerGrpc.WalletUnlockerFutureStub asyncUnlockClient = WalletUnlockerGrpc
+                .newFutureStub(LndConnection.getInstance().getSecureChannel())
+                .withCallCredentials(LndConnection.getInstance().getMacaroon());
+
+        UnlockWalletRequest asyncUnlockRequest = UnlockWalletRequest.newBuilder()
+                .setWalletPassword(ByteString.copyFrom(password, Charset.defaultCharset()))
+                .build();
+
+        final ListenableFuture<UnlockWalletResponse> unlockFuture = asyncUnlockClient.unlockWallet(asyncUnlockRequest);
+
+        unlockFuture.addListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    UnlockWalletResponse unlockResponse = unlockFuture.get();
+                    ZapLog.debug(LOG_TAG, "successfully unlocked");
+
+                    if (PrefsUtil.isWalletSetup()) {
+                        LndConnection.getInstance().stopBackgroundTasks();
+                    }
+                    LndConnection.getInstance().restartBackgroundTasks();
+
+
+                    Handler handler = new Handler();
+                    handler.postDelayed(new Runnable() {
+                        @Override
+                        public void run() {
+                            // We have to call this delayed, as without it, it will show as unconnected until the wallet button is hit again.
+                            // ToDo: Create a routine that retries this until successfull
+                            isLNDReachable();
+                        }
+                    }, 5000);
```

Comment:
So for me only 10 seconds did the trick, if I use it with 5 seconds, I run the getinfo to early, show an error and then shortly after we are connected. :/

Maybe rather set it high first and then we can improve it with the iterative approach?

Reply:
Weired... on my testsystem even 1 second was enough... I increased it to 10 for now.